contracts/AuctionHouse.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.7.3;
+
+// TODO: Use interface. The current version of @openZeppelin/contracts lib has none that fits.
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "./auctions/FungibleAuction.sol";
+
+
+/// @title Auction House
+contract AuctionHouse {
+    /// @dev All fungible auctions that are either starting in the future, already active,
+    /// or haven't had their tokens withdrawn yet after finishing.
+    address[] public fungibleAuctions;
+
+    /// @dev Create new auction for a fungible token.
+    function createFungible(
+        ERC20 sellTokenAddress,
+        ERC20 buyTokenAddress,
+        uint256 amount,
+        uint256 startTimestamp,
+        uint256 duration,
+        uint256 maxPrice,
+        uint256 minPrice
+    ) external {
+        //  liquidity checks
+        require(sellTokenAddress.balanceOf(msg.sender) >= amount, "AuctionHouse: Creator not liquid enough");
+        require(sellTokenAddress.allowance(msg.sender, address(this)) >= amount, "AuctionHouse: Not approved");
```

Comment:
wouldn't it make sense to approve if allowance not enough? Or seller has to do this explicitly?

Reply:
> Or seller has to do this explicitly?

Correct. Someone else can't call the `approve` function of an ERC20 contract for you.  See [`approve`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L130) on the template ERC20 contract, and its internal [`_approve`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L281) function.

Technically you could implement another method on your ERC20 contract that allows others to approve for you, but that's outside the scope of this PR. 😬